backend/cmd/headlamp.go

```diff
@@ -1208,3 +1209,86 @@ func absPath(path string) (string, error) {
 
 	return filepath.Join(currentUser.HomeDir, path[2:]), nil
 }
+
+//nolint:funlen
+func (c *HeadlampConfig) handleNodeDrain(w http.ResponseWriter, r *http.Request) {
+	var drainPayload struct {
+		Cluster  string `json:"cluster"`
+		NodeName string `json:"nodeName"`
+	}
+	// get nodeName and cluster from request body
+	err := json.NewDecoder(r.Body).Decode(&drainPayload)
+	if err != nil {
+		http.Error(w, "Error decoding payload", http.StatusBadRequest)
+		return
+	}
+
+	if drainPayload.NodeName == "" {
+		http.Error(w, "nodeName is required", http.StatusBadRequest)
+		return
+	}
+
+	if drainPayload.Cluster == "" {
+		http.Error(w, "clusterName is required", http.StatusBadRequest)
+		return
+	}
+	// get token from header
+	token := r.Header.Get("Authorization")
+
+	ctxtProxy := c.contextProxies[drainPayload.Cluster]
+
+	clientset, err := ctxtProxy.context.getClientSetToInteractWithKubernetesAPIServer(token)
+	if err != nil {
+		http.Error(w, "Error getting client", http.StatusInternalServerError)
+		return
+	}
+
+	nodeClient := clientset.CoreV1().Nodes()
+
+	node, err := nodeClient.Get(context.TODO(), drainPayload.NodeName, v1.GetOptions{})
+	if err != nil {
+		http.Error(w, "Error getting node", http.StatusInternalServerError)
+		return
+	}
+
+	// cordon the node first
+	node.Spec.Unschedulable = true
+	_, err = nodeClient.Update(context.TODO(), node, v1.UpdateOptions{})
+
+	if err != nil {
+		http.Error(w, "Error cordoning node", http.StatusInternalServerError)
+		return
+	}
+
+	pods, err := clientset.CoreV1().Pods("").List(context.TODO(),
+		v1.ListOptions{FieldSelector: "spec.nodeName=" + drainPayload.NodeName})
+	if err != nil {
+		panic(err)
```

Comment:
Looks like one more `panic` has to be handled.

Reply:
updated